src/google-maps/google-map/google-map.ts

```diff
@@ -17,28 +34,93 @@ import {ReplaySubject} from 'rxjs';
   template: '<div class="map-container"></div>',
 })
-export class GoogleMap implements OnInit {
-  // Arbitrarily chosen default size
-  @Input() height = '500px';
-  @Input() width = '500px';
+export class GoogleMap implements OnInit, OnDestroy {
+  @Input()
+  set height(height: string) {
+    this._height$.next(height || DEFAULT_HEIGHT);
+  }
 
-  // TODO(mbehrlich): add options, handlers, properties, and methods.
+  @Input()
+  set width(width: string) {
+    this._width$.next(width || DEFAULT_WIDTH);
+  }
 
-  private readonly _map$ = new ReplaySubject<google.maps.Map>(1);
+  @Input()
+  set center(center: google.maps.LatLngLiteral) {
+    this._center$.next(center);
+  }
+  @Input()
+  set zoom(zoom: number) {
+    this._zoom$.next(zoom);
+  }
+  @Input()
+  set options(options: google.maps.MapOptions) {
+    this._options$.next(options || DEFAULT_OPTIONS);
+  }
 
-  constructor(private readonly _elementRef: ElementRef) {}
+  // TODO(mbehrlich): Add event handlers, properties, and methods.
 
-  ngOnInit() {
-    // default options set to the Googleplex
-    const options: google.maps.MapOptions = {
-      center: {lat: 37.421995, lng: -122.084092},
-      zoom: 17,
-    };
+  private readonly _height$ = new BehaviorSubject<string>(DEFAULT_HEIGHT);
+  private readonly _width$ = new BehaviorSubject<string>(DEFAULT_WIDTH);
+  private readonly _options$ = new BehaviorSubject<google.maps.MapOptions>(DEFAULT_OPTIONS);
+  private readonly _center$ = new BehaviorSubject<google.maps.LatLngLiteral|undefined>(undefined);
+  private readonly _zoom$ = new BehaviorSubject<number|undefined>(undefined);
 
+  private readonly _destroy$ = new Subject<void>();
+
+  constructor(private readonly _elementRef: ElementRef) {
+    const googleMapsWindow: GoogleMapsWindow = window;
+    if (!googleMapsWindow.google) {
+      throw new Error(
+          'Namespace google not found, cannot construct embedded google ' +
+          'map. Please install the Google Maps JavaScript API: ' +
+          'https://developers.google.com/maps/documentation/javascript/' +
+          'tutorial#Loading_the_Maps_API');
+    }
+  }
+
+  ngOnInit() {
     const mapEl = this._elementRef.nativeElement.querySelector('.map-container');
-    mapEl.style.height = this.height;
-    mapEl.style.width = this.width;
-    const map = new google.maps.Map(mapEl, options);
-    this._map$.next(map);
+
+    const combinedOptions$: Observable<google.maps.MapOptions> =
+        combineLatest(this._options$, this._center$, this._zoom$)
+            .pipe(map(([options, center, zoom]) => {
+              const combinedOptions: google.maps.MapOptions = {
+                ...options,
+                center: center || options.center,
+                zoom: zoom !== undefined ? zoom : options.zoom,
+              };
+              if (!combinedOptions.center || !combinedOptions.center.lat ||
+                  !combinedOptions.center.lng || !combinedOptions.zoom) {
+                throw new Error(
```

Comment:
Why do we throw this error here instead of the actual maps API doing so? What does it do when it's not configured correctly?

Reply:
The map does not display and the error is displayed to the console. That actually seems fine, I've removed it.